src/device/stm32/stm32f103xx-bitfields.go

```diff
@@ -5,6 +5,10 @@
 
 package stm32
 
+// This is not used by the stm32f1xx series, but included here for the common
+// stm32 peripheral functions to operate
+type AltFunc uint8
```

Comment:
I'm not thrilled about having this here since the stm32f1xx doesn't use it, but it's needed for the common UART file as that defines a UART struct referring to this type. The alternative that I can see is to put the type-definition of the UART struct in each multiple places, which doesn't seem great either.

Reply:
The question is do the 4 lines of duplication for the type or the additional abstraction? I would probably go with the duplicate type, myself.

Reply:
Yeah that seems reasonable, and the goal was to move the code to a common place, but the types can probably live closer to their uses (i.e. at the lower level chip-specific stuff).

Reply:
Looking good now, thanks for making that change.